### DISCUSSION TITLE
curl -X 'POST api/v2/services/dhcp_server/static_mapping

### DISCUSSION THREAD

**Discussion by philippenc** [jaredhendrickson13/pfsense-api#520] [Q&A]
Hello,

When I execute a curl -X 'POST' \ https://192.168.0.1/api/v2/services/dhcp_server/static_mapping request, I get this error:
```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 327680 bytes) in /usr/local/pkg/RESTAPI/Core/Model.inc on line 373
PHP ERROR: Type: 1, File: /usr/local/pkg/RESTAPI/Core/Model.inc, Line: 373, Message: Allowed memory size of 536870912 bytes exhausted (tried to allocate 327680 bytes)
```
I increased memory_limit to 1024M in /usr/local/etc/php.ini, but it does not solve the problem. Do you have any idea what the issue might be?

Thank you.

**Comment by jaredhendrickson13** [comment]
Sounds like something might be triggering an infinite loop. Could you provide the `dhcpd` section of your config along with the request body of your POST request? I'll see if I can reproduce.

**Reply by philippenc** [reply]
Hello,

Here is the post in the response and the DHCP configuration. Additionally, I noticed in the POST request to api/v2/services/dhcp_server/static_mapping that the value of 'parent_id' cannot be an integer. As you can see in the request, the 'parent_id' is 'opt1'.

Thank you in advance for your response.

    api_url = "https://192.168.0.155/api/v2/services/dhcp_server/static_mapping"
    headers = {
        'accept': 'application/json',
        'Authorization': f'Bearer {token}',
         'Content-Type': 'application/json'}
    data = {
        "parent_id": "opt1",
        "mac": "00:50:56:b2:7f:d9",
        "ipaddr": "10.1.1.206",
        "cid": "",
        "hostname": "test-api",
        "domain": "",
        "domainsearchlist": [],
        "defaultleasetime": 7200,
        "maxleasetime": 86400,
        "gateway": "",
        "dnsserver": [],
        "winsserver": [],
        "ntpserver": [],
        "arp_table_static_entry": False,
        "descr": "machine de test pour api rest"
    }

[dhcpd-config-pfsense-restapi-dev.unc.prod-20240725131351.xml.zip](https://github.com/user-attachments/files/16369575/dhcpd-config-pfsense-restapi-dev.unc.prod-20240725131351.xml.zip)


Philippe

**Reply by jaredhendrickson13** [reply]
Thanks for providing that. I was able to reproduce the issue on v2.0.2 and verified it is a known bug that causes exponential recursion with larger datasets. This is already be fixed on the `next_minor` branch by commit 5ef2b94103f75179f0e4853960d9650566c8c89b and I've verified the issue is not present on builds from that branch. I will release a patch with the fix for this either today or tomorrow.

**Reply by philippenc** [reply]
Hello and thank you. I have another problem but I think it must be because of me. I'm opening another discussion for this.
